app/view_models/workarea/storefront/blog_view_model.rb

```diff
@@ -48,7 +48,7 @@ def page
       end
 
       def updated_at
-        @updated_at ||= model.entries.active.newest.first.try(:updated_at)
+        @updated_at ||= model.entries.select(&:active?).sort_by(&:created_at).first.try(:updated_at)
```

Comment:
It'd be better to do the sort in mongo before filtering so you don't have to do the sort in memory. You could also try grabbing a smaller sample size and grabbing the first active one so the entire collection doesn't have to load into memory.